audio/filter/af_scaletempo2.c

```diff
@@ -172,16 +172,16 @@ static bool af_scaletempo2_command(struct mp_filter *f, struct mp_filter_command
 static void af_scaletempo2_reset(struct mp_filter *f)
 {
     struct priv *p = f->priv;
-    mp_scaletempo2_reset(&p->data);
+    mp_scaletempo2_reset(p->data);
     p->initialized = false;
     TA_FREEP(&p->pending);
 }
 
 static void af_scaletempo2_destroy(struct mp_filter *f)
 {
     struct priv *p = f->priv;
-    mp_scaletempo2_destroy(&p->data);
-    talloc_free(p->pending);
+    TA_FREEP(&p->data);
```

Comment:
I'm unsure if this should be done; it's a child allocation of the filter, so it should be freed automatically. I suppose it doesn't hurt.